spec/ndx-binned-spikes.extensions.yaml

```diff
@@ -70,3 +70,41 @@ groups:
     neurodata_type_inc: DynamicTableRegion
     doc: A reference to the Units table region that contains the units of the data.
     quantity: '?'
+- neurodata_type_def: BinnedSpikes
+  neurodata_type_inc: NWBDataInterface
+  default_name: BinnedSpikes
+  doc: A data interface for non-aligned binned spike counts.
+  attributes:
+  - name: name
+    dtype: text
+    value: BinnedSpikes
+    doc: The name of this container
+  - name: description
+    dtype: text
+    value: Binned spike counts.
+    doc: A description of what the data represents
+  - name: bin_width_in_milliseconds
+    dtype: float64
+    doc: The length in milliseconds of the bins
+  - name: milliseconds_from_event_to_first_bin
+    dtype: float64
+    default_value: 0.0
+    doc: The time in milliseconds from the event to the beginning of the first bin.
+      A negative value indicatesthat the first bin is before the event whereas a positive
+      value indicates that the first bin is after the event.
+    required: false
```

Comment:
What is the use case for this? I think the most common use case for this type is that the first bin starts at the session start time and runs to the end of the session, so this field is not needed. 

But maybe the experiment doesn't start until time t and so the user cares only about spikes after time t. If that is a common case, then I think there should be a field that says what the starting event is. If there is a starting event, should there also be an ending event?

Reply:
Ah, so this is meant to cover the whole experiment then. So it would have a large number of bins usually. Is that correct?

In that case I agree this does not make sense. We should just have a `start_time` that is 0 by default, what do you think?

Reply:
In general, I [worry](https://github.com/hdmf-dev/hdmf-schema-language/issues/45) about setting default values in the schema because a data reuser does not know whether that was explicitly set or automatically and unintentionally set because it was the default. I prefer things to be explicit. So I would make `start_time` required with no default.

Reply:
We can always add the default value later based on feedback.

Reply:
I'm concerned that adding more requirements could make the API harder to use, especially when it comes to instantiating data.

I’ve read your issue and I'm thinking it through. In this specific case, since we're dealing with floats, we can use `np.nan` as a sentinel value. It clearly indicates that the information is unavailable, so anyone retrieving the data will understand that right away. Another option is to make the field `Optional` without a default value, allowing its absence to convey the same meaning.

What do you think?

Reply:
Circling back to this, I think we decided that `default_value: 0.0` for an optional field is OK, and that the APIs need to be updated so that a user can reset the value to null (https://github.com/NeurodataWithoutBorders/pynwb/issues/2105).

It's still not clear whether we really need this field but I am OK with keeping it.